install/kubernetes/helm/istio/charts/mixer/templates/create-config-job.yaml

```diff
@@ -0,0 +1,41 @@
+{{- $istio_namespace := .Values.global.namespace }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  annotations:
+    "helm.sh/hook": post-install
```

Comment:
I am a bit confused about logic here, after reading helm hooks docs, I thought we should load all dependency as pre-install hooks with weights to order them, so namespace creation would go first, then RBACs then CRD etc.. 
Here you use post-install, please comment why.

Reply:
issue is that there is a race condition with CDRs, you can install them as `pre-install` hook but that doesn't warranty that they'll be immediately available to consume, so that's why I install them normally and then trigger a `post-install` hook (`get-crd-job.yaml`) that only waits till all CRDs are registered on the api and available to consume (the job only does `kubectl get crd ...`). After that a second `post-install` hook is triggered (`create-config-job`), which install the objects that depends on the previous CRD.

Reply:
https://github.com/kubernetes/helm/blob/master/docs/charts_hooks.md
What about using --wait keyword, as it is described in steps 6. 7. and 8. Unless there is a bug in helm's tiller implementation I do not see why it would not work with pre-install and --wait.

Reply:
it only wait till installation is successful, but that doesn't mean the crd is ready to consume (it pass some time before it is registered in the api by the CRD controller, I think :) will read more about this), I tried it before and helm still complained about the ""config.istio.io" not being defined. But maybe I did something wrong, will try again.

Reply:
@sbezverk https://github.com/kubernetes/helm/issues/2925#issuecomment-328904829

Reply:
--wait seems a good choice here if it works.

Reply:
@luisyonaldo nice work here.  Your last comment about crd being ready is accurate.  Nearly all wait operations wait until the registration has been started, not that it has been completed.  Your approach here looks sound.